### PR TITLE
Gauge integration

### PR DESCRIPTION
**Pull request by monofrei**

### CODE REVIEW

File: gauge-tests/.gauge/runInfo.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+		"run",
+		"--machine-readable",
+		"--hide-suggestion",
+		"--simple-console",
+		"--env",
+		"chrome"
+	],
+	"FailedItems": [],
+	"Items": [
+		"D:/REPO/koshmar/gauge-tests/specs/checkBoxesComponentTests.spec"
```
Comment (costea32) [inline gauge-tests/.gauge/runInfo.json]: This doesn't look right :)

File: gauge-tests/env/chrome/browser.properties [hunk 2]
```diff
@@ -0,0 +1,2 @@
+browser = chrome
+webdriver.chrome.driver =D:/WebDriver/chromedriver/chromedriver23.exe
```
Comment (costea32) [inline gauge-tests/env/chrome/browser.properties]: paths... can we avoid that?

File: gauge-tests/properties/project.properties [hunk 3]
```diff
@@ -0,0 +1 @@
+webdriver.chrome.driver =D:/WebDriver/chromedriver/chromedriver23.exe
```
Comment (costea32) [inline gauge-tests/properties/project.properties]: the same property was defined somewhere before

File: gauge-tests/src/test/java/org/testmonkeys/tests/gauge/ScenarioContext.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        }
+        browserType = System.getenv("browser");
+        URL url = this.getClass().getClassLoader().getResource("WebPages");
+        if (url == null){
+            throw new RuntimeException("Static pages folder not found");
+        }
+        this.baseUrl = new File(url.getFile()).getAbsolutePath();
+    }
+
+    public static ScenarioContext getInstance() {
```
Comment (costea32) [inline gauge-tests/src/test/java/org/testmonkeys/tests/gauge/ScenarioContext.java]: should be synchronized

File: gauge-tests/src/test/java/org/testmonkeys/tests/gauge/ScenarioContext.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        if (instance == null) {
+            instance = new ScenarioContext();
+        }
+
+        return instance;
+    }
+
+
+    public void reset() {
```
Comment (costea32) [inline gauge-tests/src/test/java/org/testmonkeys/tests/gauge/ScenarioContext.java]: empty method

File: koshmar/src/main/java/org/testmonkeys/koshmar/core/factory/PageScanner.java [hunk 6]
```diff
@@ -38,4 +49,18 @@ private Supplier pageWithNameNotFound(String name) {
         Optional<Class<?>> first = typesAnnotatedWith.stream().filter(isPageWithName(name)).findFirst();
         return first.orElseThrow(pageWithNameNotFound(name));
     }
+
+    public <T extends AbstractComponent> T findPageElementByName(Page page, String name) {
```
Comment (costea32) [inline koshmar/src/main/java/org/testmonkeys/koshmar/core/factory/PageScanner.java]: The class is mostly used to scan the packages for page objects classes, method, by it's name seems to need a separate class

File: gauge-tests/.gauge/runInfo.json (hunk 1 above)
Comment (monofrei) [reply gauge-tests/.gauge/runInfo.json]: missed it from ignore

File: gauge-tests/env/chrome/browser.properties (hunk 2 above)
Comment (monofrei) [reply gauge-tests/env/chrome/browser.properties]: i will put a note there, couldn't find how to run gauge with env variables, it is some witchcraft there.

File: gauge-tests/properties/project.properties (hunk 3 above)
Comment (monofrei) [reply gauge-tests/properties/project.properties]: will remove

File: gauge-tests/src/test/java/org/testmonkeys/tests/gauge/ScenarioContext.java (hunk 4 above)
Comment (monofrei) [reply gauge-tests/src/test/java/org/testmonkeys/tests/gauge/ScenarioContext.java]: will be done

**Pull request merged**